### ISSUE TITLE
Problem receiving binary messages

### ISSUE THREAD

**Issue by huangqingbiao8831** [Theldus/wsServer#18]
The binary message sent is：0x1a,0x2b,0x3c,0x4d,0x00,0x5c,0x6d,0xdd,0xaa,0xac ，But received as ：0x1a ,0x2b ,0x3c ,0x4d ,0x3f  ,0xe5 ,0x32 ,0xa2 ,0x95 ,0x15
If it does not contain 00 bytes when sending, the received message is correct, could you give me some suggestions ，thanks .

**Comment by Theldus** [comment]
Hi @huangqingbiao8831,
> The binary message sent is: [....], But received as [....]

Ok, but in which direction? from the client (browser, for example) to wsServer or from wsServer to client?

**a) If the former (client -> wsServer):**
I did some tests with a hand-crafted frame with the same content that you passed and it worked as expected, wsServer is able to correctly receive the frame and display the bytes on the screen (with some modifications in example/send_receive.c).

If you want to reproduce, launch wsServer in one terminal and in another, run:
```bash
printf  "\x47\x45\x54\x20\x2f\x20\x48\x54\x54\x50\x2f\x31\x2e\x31\x0d\x0a"\
	"\x48\x6f\x73\x74\x3a\x20\x31\x32\x37\x2e\x30\x2e\x30\x2e\x31\x3a\x38"\
	"\x30\x38\x30\x0d\x0a\x43\x6f\x6e\x6e\x65\x63\x74\x69\x6f\x6e\x3a\x20"\
	"\x55\x70\x67\x72\x61\x64\x65\x0d\x0a\x55\x70\x67\x72\x61\x64\x65\x3a"\
	"\x20\x77\x65\x62\x73\x6f\x63\x6b\x65\x74\x0d\x0a\x53\x65\x63\x2d\x57"\
	"\x65\x62\x53\x6f\x63\x6b\x65\x74\x2d\x56\x65\x72\x73\x69\x6f\x6e\x3a"\
	"\x20\x31\x33\x0d\x0a\x53\x65\x63\x2d\x57\x65\x62\x53\x6f\x63\x6b\x65"\
	"\x74\x2d\x4b\x65\x79\x3a\x20\x53\x67\x5a\x77\x71\x57\x68\x57\x78\x65"\
	"\x59\x44\x78\x76\x56\x64\x63\x55\x6a\x4a\x49\x51\x3d\x3d\x0d\x0a\x0d"\
	"\x0a\x82\x8a\x0b\x0b\x0b\x0b\x11\x20\x37\x46\x0b\x57\x66\xd6\xa1\xa7"\
	"\x88\x82\xfb\xf1\x90\xa7\xf8\x18" | nc 127.0.0.1 8080
```
The above command sends a handshake frame, followed by a 10-byte frame (masked, with mask "0b 0b 0b 0b") with the contents 0x11, 0x20, 0x37, 0x46, 0x0b, 0x57, 0x66, 0xd6, 0xa1, 0xa7 (the same informed by you, but masked), followed by a close frame.

If you modify the `on_message()` method in the example/send_receive.c file to something like:
```c
printf("Message: ");
for (uint64_t i = 0; i < size; i++)
	printf("%02x ", msg[i]);
printf("\n");
```

You will get the following output:
```bash
$ ./example/send_receive
Waiting for incoming connections...
Connection opened, client: 4 | addr: 127.0.0.1
Message: 1a 2b 3c 4d 00 5c 6d dd aa ac
```

**b) If the latter (wsServer -> client)):**
If you want to actually send binary frames **from** wsServer, you should use the `ws_sendframe_bin` function, not `ws_sendframe_txt`. Something like this should work:

```c
unsigned char buff[] = {0x1a, 0x2b, 0x3c, 0x4d, 0x00, 0x5c, 0x6d, 0xdd, 0xaa, 0xac};
ws_sendframe_bin(fd, buff, sizeof(buff), 0);
```
---
TL;DR: I believe you are using the `ws_sendframe_txt` function, which expects a string and thus considers the null byte as the string delimiter. Using ws_sendframe_bin should solve your problem.

Any questions, let me know. All available functions are documented online [here](https://theldus.github.io/wsServer/include_2ws_8h.html#aec4fce2b51671706c217e84721f20856).

**Comment by huangqingbiao8831** [comment]
I will try again, thank you for your patience and advice。
